app/src/main/java/org/mozilla/fenix/perf/ProfilerDialogFragment.kt

```diff
@@ -0,0 +1,455 @@
+                            fontWeight = FontWeight.ExtraBold,
+                            fontSize = 20.sp,
+                            modifier = Modifier.padding(8.dp)
+                        )
+                        Text(
+                            text = stringResource(R.string.profiler_settings_title),
+                            fontWeight = FontWeight.Bold,
+                            fontSize = 15.sp,
+                            modifier = Modifier.padding(8.dp)
+                        )
+                        Spacer(modifier = Modifier.height(2.dp))
+                        LabeledRadioButton(
+                            text = stringResource(R.string.profiler_filter_firefox),
+                            subText = stringResource(R.string.profiler_filter_firefox_explain),
+                            state = profilerSettingsObserver
+                        )
+
+                        LabeledRadioButton(
+                            text = stringResource(R.string.profiler_filter_graphics),
+                            subText = stringResource(R.string.profiler_filter_graphics_explain),
+                            state =  profilerSettingsObserver
+                        )
+
+                        LabeledRadioButton(
+                            text = stringResource(R.string.profiler_filter_media),
+                            subText = stringResource(R.string.profiler_filter_media_explain),
+                            state = profilerSettingsObserver
+                        )
+
+                        LabeledRadioButton(
+                            text = stringResource(R.string.profiler_filter_networking),
+                            subText = stringResource(R.string.profiler_filter_networking_explain),
+                            state = profilerSettingsObserver
+                        )
+                        Spacer(modifier = Modifier.height(8.dp))
+                        Row(
+                            horizontalArrangement = Arrangement.End,
+                            modifier = Modifier.fillMaxWidth()
+                        ) {
+                            TextButton(onClick = {
+                                openDialog.value = false
+                                this@ProfilerDialogFragment.dismiss()
+                            }) {
+                                Text(text = stringResource(R.string.profiler_start_cancel))
+                            }
+                            Spacer(modifier = Modifier.width(4.dp))
+                            TextButton(onClick = {
+
+                                openDialog.value = false
+                                val settingType = ProfilerSettings.valueOf(profilerSettingsObserver.value)
+                                onPositiveClick(settingType.threads!!, settingType.features!!)
+                                profilerActiveObserver.value = true
+
+                                Thread{
```

Comment:
thought: we have to make sure this click event is debounced so it doesn't get called twice and cause too many jobs to run.